sthgetter/sthgetter.go

```diff
@@ -68,7 +72,72 @@ func getCheckStoreSTH(ctx context.Context, lc *client.LogClient, st storage.APIC
 		glog.Infof("%s: %s: error writing API Call %s: %s", l.URL, logStr, apiCall, err)
 	}
 
-	//TODO(katjoyce): Run checks on the received STH.
+	if sth == nil {
+		return
+	}
+
+	// Verify the STH.
+	errs := checkSTH(sth, apiCall.End, sv, l)
+	if len(errs) != 0 {
+		var b strings.Builder
+		fmt.Fprintf(&b, "STH verification errors for STH %v:", sth)
+		for _, e := range errs {
+			fmt.Fprintf(&b, "\n\t%v: %v,", reflect.TypeOf(e), e)
+		}
+		glog.Info(b.String())
+	}
+
+	// TODO(katjoyce): Store STH & errs.
+}
+
+func checkSTH(sth *ct.SignedTreeHead, receivedAt time.Time, sv *ct.SignatureVerifier, l *ctlog.Log) []error {
+	var errs []error
+	// Check that the STH signature verifies.
+	log.Printf("%s: %s: verifying STH signature...", l.URL, logStr)
+	if err := sv.VerifySTHSignature(*sth); err != nil {
+		errs = append(errs, &SignatureVerificationError{Err: err})
+		log.Printf("%s: %s: STH signature verification failed", l.URL, logStr)
+	}
+
+	// Check STH is not older than the MMD of the Log.
+	if err := checkSTHTimestamp(sth, receivedAt, l.MMD); err != nil {
+		errs = append(errs, &OldTimestampError{Err: err})
+		log.Printf("%s: %s: STH timestamp verification failed", l.URL, logStr)
+	}
+
+	// TODO(katjoyce): Implement other checks on the STH:
```

Comment:
Could also check that the root hash is the correct length (i.e. a valid hash).

Reply:
I believe this is already checked in the client when converting from GetSTHResponse to SignedTreeHead (see http://go/gh/google/certificate-transparency-go/blob/master/types.go#L487).

Reply:
Might be worth adding a test that covers this then, since it's an external bit of code you're counting on not changing.

Reply:
Ah you don't have any tests for `Run()` yet, which would be where you'd want to cover this case. That can be deferred then.

Reply:
Having chatted offline, I agree, once we have tests for Run() in place (which we don't yet).